gems/sorbet-runtime/lib/types/props/decorator.rb

```diff
@@ -277,6 +277,22 @@ def prop_validate_definition!(name, cls, rules, type)
     end
   end
 
+  # Returns `true` when the type of the prop is nilable, or a `:default` is
+  # present in the rules hash, and its value is `nil`.
+  #
+  # checked(:never) - Rules hash is expensive to check
+  sig do
+    params(
+      cls: PropTypeOrClass,
+      rules: Rules,
+    )
+    .void
+    .checked(:never)
+  end
+  private def prop_nilable?(cls, rules)
+    T::Utils::Nilable.is_union_with_nilclass(cls) || (rules.key?(:default) && rules[:default].nil?)
+  end
```

Comment:
Do we have tests for something like `prop :thing, Integer, default: nil`?  (I guess we'd want both Sorbet checks, if possible, and sorbet-runtime checks?)  The second part of the `||` is really only relevant for untyped `cls`, right?